docs/C_API.md

```diff
@@ -39,5 +39,18 @@ The example below shows a sample run using the SqueezeNet model from ONNX model
 
 * [../csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/C_Api_Sample.cpp](../csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/C_Api_Sample.cpp)
 
+## Deployment
+
+### Windows 10
+
+Your installer should put the onnxruntime.dll into the same folder as your application.   Your application can either use [load-time dynamic linking](https://docs.microsoft.com/en-us/windows/win32/dlls/using-load-time-dynamic-linking) or [run-time dynamic linking](https://docs.microsoft.com/en-us/windows/win32/dlls/using-run-time-dynamic-linking) to bind to the dll.
+
+#### Dynamic Link Library Search Order
+
+This is an important article on how Windows finds supporting dlls: [Dynamic Link Library Search Order](https://docs.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order).
+
+There are some cases where a middleware dll is using the onnxruntime and is not in the same folder as the app.  Do not modify the system %path% variable to add your middleware folder.  This can conflict with other software on the machine that is also using the onnxruntme.  Instead you must use [run-time dynamic linking](https://docs.microsoft.com/en-us/windows/win32/dlls/using-run-time-dynamic-linking) to bind explicity to the copy in your folder.  You can use code like this sample does in [GetModulePath()](https://github.com/microsoft/Windows-Machine-Learning/blob/master/Samples/SampleSharedLib/SampleSharedLib/FileHelper.cpp) to find out what folder your dll is loaded from.  You use that to build the path to the onnxruntime.dll that you pass to LoadLibrary().
```

Comment:
Should we clarify what middleware means here?

Reply:
it's a good point.  i wasn't sure about using that word or not.  ideas on other things to call it?  maybe toolkits?  engines?  runtimes?    or just keep it generic like "where another DLL is using the onnxruntime DLL"

Reply:
made a change.   stopped using middleware.   what do you think ?